### PR TITLE
[Enhancement][Regression-Test] enhancement for regression-test framework

### LINKED ISSUE #8729: [Enhancement] enhancement for regression-test framework
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

1. support double type relative-error check.
2. adapt diffrent datetime format.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes
@924060929 PTAL
Issue Number: close #8729

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/util/OutputUtils.groovy [hunk 1]
```diff
@@ -44,9 +53,36 @@ class OutputUtils {
         return writer.toString()
     }
 
+    static String checkCell(String info, int line, String expect_cell, String real_cell, String data_type) {
+        if(data_type == "FLOAT" || data_type == "DOUBLE") {
```
Comment (924060929) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/util/OutputUtils.groovy:57]: suggest change to camel case

File: regression-test/framework/src/main/groovy/org/apache/doris/regression/util/JdbcUtils.groovy [hunk 2]
```diff
@@ -18,37 +18,40 @@
...
+import org.apache.doris.regression.util.Metaholder
 
 import java.sql.Connection
 import java.sql.ResultSet
+import java.sql.ResultSetMetaData
 
 class JdbcUtils {
-    static List<List<Object>> executeToList(Connection conn, String sql) {
+    static List<List<Object>> executeToList(Connection conn, String sql, Metaholder holder) {
         conn.prepareStatement(sql).withCloseable { stmt ->
```
Comment (924060929) [inline regression-test/framework/src/main/groovy/org/apache/doris/regression/util/JdbcUtils.groovy:29]: suggest `new Metaholder` in `JdbcUtils` and return, and you can extract the interest metadata out of the `Connection` life cycle safely.

e.g.

```groovy
class Metadata {
  public final List<String> columnNames;
  public final List<String> columnTypes;
}

static Tuple2<List<List<Object>>, Metadata> executeToList(Connection conn, String sql) {
  def metadata = new Metadata()
  resultList = ...
  metadata.columnNames = ...
  metadata.columnTypes = ...
  return [resultList, metadata]
}
```
then, you can take result by this code
```groovy
def (result, metadata) = JdbcUtils.executeToList(connnection, sql)
```

Comment (BiteTheDDDDt) [reply regression-test/framework/src/main/groovy/org/apache/doris/regression/util/JdbcUtils.groovy:29]: fixed

**Comment by 924060929** [comment]
+1

**Comment by 924060929** [comment]
+1

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**